File: app/src/main/java/im/tox/antox/callbacks/AntoxOnReadReceiptCallback.java
package im.tox.antox.callbacks;

import android.content.Context;

import im.tox.antox.data.AntoxDB;
import im.tox.antox.tox.ToxSingleton;
import im.tox.antox.utils.AntoxFriend;
import im.tox.jtoxcore.callbacks.OnReadReceiptCallback;

public class AntoxOnReadReceiptCallback implements OnReadReceiptCallback<AntoxFriend> {

    private final static String TAG = "im.tox.antox.TAG";
    private Context ctx;
    ToxSingleton toxSingleton = ToxSingleton.getInstance();

    public AntoxOnReadReceiptCallback(Context ctx) {
        this.ctx = ctx;
    }

    @Override
    public void execute(AntoxFriend friend, int receipt) {
        AntoxDB db = new AntoxDB(this.ctx);
        String key = db.setMessageReceived(receipt);
        db.close();

        /* Broadcast */
        toxSingleton.updateMessages(ctx);
    }
}

File: app/src/main/scala/im/tox/antox/callbacks/AntoxOnReadReceiptCallback.scala
package im.tox.antox.callbacks

import android.content.Context
import android.util.Log
import im.tox.antox.callbacks.AntoxOnReadReceiptCallback._
import im.tox.antox.data.{State, AntoxDB}
import im.tox.antox.tox.ToxSingleton
import im.tox.tox4j.core.callbacks.FriendReadReceiptCallback

object AntoxOnReadReceiptCallback {

  private val TAG = "im.tox.antox.callbacks.AntoxOnReadReceiptCallback"
}

class AntoxOnReadReceiptCallback(private var ctx: Context) extends FriendReadReceiptCallback[Unit] {
  override def friendReadReceipt(friendNumber: Int, messageId: Int)(state: Unit): Unit = {
    val db = State.db
    db.setMessageReceived(messageId)
    Log.d(TAG, "read receipt, for message " + messageId)
  }
}
